js-library/src/utils/keys.ts

```diff
@@ -0,0 +1,41 @@
+import { utils } from 'ethers'
+import { isHexString } from 'ethers/lib/utils'
+import {
+  HexString,
+  PublicKey,
+  PublicKeyX,
+  PublicKeyY,
+  PUBLIC_KEY_LENGTH,
+  PUBLIC_KEY_PART_LENGTH,
+} from '../model/hex.types'
+
+const { hexStripZeros } = utils
+
+export function isHexStringPublicKey(hexString: HexString): boolean {
+  return isHexString(hexString) && hexString.length === PUBLIC_KEY_LENGTH
+}
+
+export function isHexStringPublicKeyPart(hexString: HexString): boolean {
+  return isHexString(hexString) && hexString.length === PUBLIC_KEY_PART_LENGTH
+}
+
+export function isPublicKeyValid(publicKey: PublicKey): boolean {
```

Comment:
as I meantioned use user-defined type guards for these problems
```suggestion
export function isPublicKeyValid(publicKey: unknown): publicKey is PublicKey {
```

Reply:
please correct it at other places as well